app/models/spree/variant_decorator.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+# Flow specific methods for Spree::Variant
+module Spree
+  Variant.class_eval do
+    def flow_stock?(quantity)
```

Comment:
@texpert I was in doubt related to this method. I was going to reuse the `Spree::Variant#available_online?` method but since I needed to change it to support receiving the quantity as parameter I decided first to add this method instead. 

What do you think? If you agree I can add a new commit on `feature/tec-3339-flow_io_technical_assessment` changing this `available_online?` method to allow receiving a quantity and then update this code to reuse that method.

Reply:
Yeah, I think it is better to reuse/override the `Spree::Variant#available_online?` - sure you can push a commit to flow branch.

Please, note, though, that flowcommerce_spree is decorating Spree::Variant using module prepending technique - https://github.com/mejuri-inc/flowcommerce_spree/blob/develop/app/models/spree/flow_io_variant_decorator.rb

This has the advantage to ensure that if a method with the same name is defined in Spree gem, then overridden in mejuri-web by class_eval, and then overridden in flowcommerce_spree gem, the latter will be founf first by the Ruby method lookup.

I've tried first with class_eval and it hadn't work - it was putting the flowcomeerce_spree module between the Spree gem and the mejuri-web class, so, the similarly named methods were ignored.

So, please, move this code into Spree::FlowIoVariantDecorator

Reply:
Great! I will push the change related to `Spree::Variant#available_online?` I will have into account the prepending technique for the future too! thanks for the whole explanation!